lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -80,22 +80,15 @@ static CanType getSingleElementDeclFieldType(NominalTypeDecl *decl) {
 /// Given a generic function with a single generic parameter `<T>(...) -> ...`
 /// an a type for substitution, return a substitution map that's suitable for
 /// calling this func.
-static SubstitutionMap getSingleSubstitutionMap(SILFunction *f, Type ty) {
-  return SubstitutionMap::get(
-    f->getGenericEnvironment()->getGenericSignature(),
-    [&](SubstitutableType *t) { return ty; },
-    MakeAbstractConformanceForGenericType());
-}
-
-/// Given a known builtin name and a type for substitution, returns a
-/// substitution map suitable for calling this builtin.
-static SubstitutionMap getSingleSubstitutionMapForBuiltin(
-  Identifier builtinName, Type ty, ASTContext &ctx) {
-  auto builder = GenericSignatureBuilder(ctx);
-  builder.addGenericParameter(GenericTypeParamType::get(0, 0, ctx));
-  auto *genSig = std::move(builder).computeGenericSignature(SourceLoc());
-  return SubstitutionMap::get(genSig, [&](SubstitutableType *t) { return ty; },
-                              MakeAbstractConformanceForGenericType());
+static SubstitutionMap
+getSingleSubstitutionMapForFunction(SILFunction *f, Type ty,
+                                    SILModule &userModule) {
+  auto *loadedFunc = lookupOrLinkFunction(f->getName(), userModule);
+  assert(loadedFunc->getGenericEnvironment());
+  auto *genericSig = loadedFunc->getGenericEnvironment()->getGenericSignature();
```

Comment:
Assert that genericSig->getGenericParams().size() is 1?

Reply:
Done.